packages/core/src/eval/evaluation.ts

```diff
@@ -0,0 +1,22 @@
+import { type Agent } from '../agent';
+import { AvailableHooks, executeHook } from '../hooks';
+
+import { type Metric } from './metric';
+
+export async function evaluate<T extends Agent>(agent: T, input: Parameters<T['generate']>[0], metric: Metric) {
+  const agentOutput = await agent.generate(input);
```

Comment:
@abhiaiyer91 you were talking about options to test streams etc as well? I can create multiple evaluate functions or  perhaps something like
```
 evaluate((input) => electionAgent.generate(input), 'Who won the 2016 US presidential election?', new TestMetric());
```